spring-integration-sftp/src/main/java/org/springframework/integration/sftp/session/DefaultSftpSessionFactory.java

```diff
@@ -191,8 +193,21 @@ public void setPassword(String password) {
 	 * false (default).</b>
 	 * @param knownHosts The known hosts.
 	 * @see JSch#setKnownHosts(String)
+	 * @deprecated since 5.2.5 in favor of {@link #setKnownHostsResource(Resource)}
 	 */
+	@Deprecated
```

Comment:
Should we really deprecate? This is easier if you are using Java Config and want to point to a simple file name.

Not related to this PR but perhaps we should consider adding a `SftpSessionFactorySpec` ?

Reply:
Then we need also to add a string-based for `privateKey`.

More over it is better to resolve any resources via `@Value Resource` then inject.